roles/cs.aws-ec2-cleanup/tasks/main.yml

```diff
@@ -1,60 +1,130 @@
+- name: Get available Launch Configurations (TEMPORARY STEP)
+  ec2_lc_info:
+    region: "{{ aws_region }}"
+  register: _aws_ec2_cleanup_all_lcs
 
-- debug:
-    msg: "Keeping {{ _aws_ec2_cleanup_ami_ids_to_keep | count }} AMIs: {{ _aws_ec2_cleanup_ami_ids_to_keep | join(', ') }}"
+- name: Keep amis required for all launch configurations (TEMPORARY STEP)
+  set_fact:
+    _ami_to_keep: "{{ _ami_to_keep + [item] }}"
+  when: item not in _ami_to_keep
+  loop: "{{ _aws_ec2_cleanup_all_lcs.launch_configurations | map(attribute='image_id') | list}}"
+
+- name: Select versions to remove
+  set_fact:
+    _remove_template_version: "{{ _remove_template_version + [_version] }}"
+  vars:
+    _version:
+      name: "{{ item.LaunchTemplateName }}"
+      version: "{{ item.VersionNumber }}"
+  when: item not in _to_keep
+  loop: "{{ _launch_template_version_output.stdout_lines | map('from_json') | map(attribute='LaunchTemplateVersions') | flatten }}"
 
 - name: Get list of App AMIs
   ec2_ami_info:
     region: "{{ aws_region }}"
     owner: self
-    filters: "{{ aws_tags_base | prefix_keys('tag:') | combine(aws_tags_role_app_filter) }}"
-  vars:
-    aws_tags_role_app_filter: "{{ aws_tags_role_app | prefix_keys('tag:') }}"
+    filters: "{{ aws_tags_default | combine(aws_tags_role_app_base) | prefix_keys('tag:')}}"
   register: _aws_ec2_cleanup_env_amis
 
-- name: "Remove Launch Configurations not longer required"
-  ec2_lc:
-     region: "{{ aws_region }}"
-     state: absent
-     name: "{{ item.launch_configuration_name }}"
-  when: item.launch_configuration_name not in _aws_ec2_cleanup_lc_ids_to_keep
-  with_items: "{{ _aws_ec2_cleanup_env_lcs }}"
+- name: Cleanup lists
+  set_fact:
+    _to_keep: "{{ _to_keep | unique }}"
+    _ami_to_keep: "{{ _ami_to_keep | unique }}"
+    _amis_found: "{{ _aws_ec2_cleanup_env_amis.images | map(attribute='image_id') | list }}"
+
+- name: Select amis to remove
+  set_fact:
+    _amis_to_remove: "{{ _amis_found | difference(_ami_to_keep) }}"
+
+- name: List launch templates to remove
+  debug:
+    msg: |
+      Launch template versions to keep:
+      {% for item in _to_keep %}
+        {{ item.LaunchTemplateName }}:{{ item.VersionNumber }} {% if item.DefaultVersion %}(Default){% endif %}
 
-- name: "Remove AMIs not longer required"
+      {% endfor %}
+
+      Launch template versions to remove
+      {% for item in _remove_template_version %}
+        {{ item.name }}:{{ item.version }}
+      {% endfor %}
+
+      Amis to keep:
+      {% for item in _ami_to_keep %}
+        {{ item }}
+      {% endfor %}
+
+      Amis to remove:
+      {% for item in _amis_to_remove %}
+        {{ item }}
+      {% endfor %}
+
+- name: Remove obsoled LT version
+  command:
+    cmd: "aws ec2 delete-launch-template-versions --launch-template-name {{ item.name }} --versions {{ item.version }} --region {{ aws_region }}"
```

Comment:
quote LT name with `{{ item.name | quote }}`  to ensure there is no code injection at this point